jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/sessionconfig/CodeScanSessionConfig.kt

```diff
@@ -116,6 +116,13 @@ internal sealed class CodeScanSessionConfig(
         false -> "${getPayloadLimitInBytes() / TOTAL_BYTES_IN_KB}KB"
     }
 
+    open fun getTotalProjectSizeInBytes(): Long {
+        fun getDirSize(file: VirtualFile): Long = file.children.fold(0L) { acc, next ->
+            acc + if (next.isDirectory) getDirSize(next) else next.length
+        }
+        return getDirSize(projectRoot)
+    }
+
```

Comment:
Does this handle edge cases properly? (e.g. symlinks)

Reply:
Updated to handle symlinks. Also added tests covering the method.